android/modules/network/src/java/ti/modules/titanium/network/TiHTTPClient.java

```diff
@@ -1271,12 +1271,12 @@ protected void setUpClient(HttpURLConnection client, Boolean isPostOrPutOrPatch)
 		        return;
 		    }
 
-		    client.setUseCaches(true);
 		    client.setRequestMethod(method);
 		    client.setDoInput(true);
 
 		    if (isPostOrPutOrPatch) {
 		        client.setDoOutput(true);
+		        client.setChunkedStreamingMode(0);
```

Comment:
I did initially want to use `setFixedLengthStreamingMode` but `totalLength` isn't the true length of the content that gets sent. The data gets manipulated into a form, which causes length issues when using `setFixedLengthStreamingMode`.

I did test against HTTPS and here's one with a redirect that works too:
```Javascript
var http = Ti.Network.createHTTPClient({
		onload: function() {
			Ti.API.info('response: ' + this.responseText);
		}
	});
http.open('POST', 'https://jigsaw.w3.org/HTTP/300/301.html');
http.send();
```